fix markdown and nextflow_schema.json check

yuukiiwa opened the pull request:
hi @cying111 , I have updated usage.md with more specific headers and added the missing 34 parameters to nextflow_schema.json to avoid errors.

yuukiiwa committed:
more specific headers for passing markdownlint

yuukiiwa committed:
Update nextflow_schema.json

cying111 commented on docs/usage.md:
```diff
@@ -30,7 +30,7 @@ As shown in the examples below, the accepted format of the file is slightly diff
 #### With basecalling and demultiplexing
 
-##### Example `samplesheet.csv`
+##### Example `samplesheet.csv` for barcoded fast5 inputs
```
any reason why you add this? Same for below

yuukiiwa replied:
I got the following error from the markdownlint check:
![Screenshot 2020-08-14 at 4 40 18 PM](https://user-images.githubusercontent.com/41866052/90230926-e7705180-de4c-11ea-9c9f-487f0b4fd766.png)

yuukiiwa replied:
there's this no duplicate heading requirement

yuukiiwa committed:
update header

yuukiiwa committed:
Update runDESeq2.R

yuukiiwa committed:
Update runDEXseq.R

yuukiiwa committed:
add in Harshil's updates to solve merging conflict

yuukiiwa committed:
add in Harshil's update to solve merge conflict

yuukiiwa committed:
re-update README with new changes

yuukiiwa committed:
check params.input_path

yuukiiwa committed:
check value from --input_path

yuukiiwa committed:
add test for bam file input

yuukiiwa committed:
resolve merge conflict with nf-core/nanoseq

yuukiiwa committed:
reupdate after solving merge conflict with nf-core/nanoseq

yuukiiwa commented:
hi @cying111, I made some changes and ran through the test profiles locally with the updated test-datasets:
test,docker
![test](https://user-images.githubusercontent.com/41866052/90611994-72e33d00-e1d5-11ea-9246-cde7e18622d8.png)
test_bc_nodx,docker
![test_bc_nodx](https://user-images.githubusercontent.com/41866052/90612073-89899400-e1d5-11ea-948f-8d9efcc54ae4.png)
test_nobc_dx,docker
![test_nobc_dx](https://user-images.githubusercontent.com/41866052/90612081-8d1d1b00-e1d5-11ea-9937-9ee32edb1304.png)
test_nobc_nodx,docker
![test_nobc_nodx](https://user-images.githubusercontent.com/41866052/90612093-90b0a200-e1d5-11ea-8b26-b7f887f5236c.png)
test_nobc_nodx_noaln
![test_nobc_nodx_noaln](https://user-images.githubusercontent.com/41866052/90612104-94442900-e1d5-11ea-98ba-b4da7ee3d848.png)

yuukiiwa committed:
add missing parameters

yuukiiwa committed:
swap input_file and barcode columns

yuukiiwa commented:
hi @cying111, I updated the test-dataset samplesheets and all tests ran through:
test,docker
![test](https://user-images.githubusercontent.com/41866052/90619317-19800b80-e1df-11ea-8a65-65f28dbd77c3.png)
test_bc_nodx,docker
![test_bc_nodx](https://user-images.githubusercontent.com/41866052/90619329-1c7afc00-e1df-11ea-8750-a52dd443d38c.png)
test_nobc_dx,docker
![test_nobc_dx](https://user-images.githubusercontent.com/41866052/90619344-200e8300-e1df-11ea-9e65-678477613cd9.png)
test_nobc_nodx,docker
![test_nobc_nodx](https://user-images.githubusercontent.com/41866052/90619355-23a20a00-e1df-11ea-8b6b-c97c1223c6d3.png)
test_nobc_nodx_noaln,docker
![test_nobc_nodx_noaln](https://user-images.githubusercontent.com/41866052/90619374-2997eb00-e1df-11ea-83c0-172a22d3e976.png)

yuukiiwa committed:
check "group,replicate,barcode,input_file" columns

- use group column to indicate sample in --input_path is provided
- use input_file column to point to the path to samples if --input_path is not provided

yuukiiwa commented:
the tests ran through locally with the updated samplesheet format:
test,docker
![test](https://user-images.githubusercontent.com/41866052/90712197-87c0de80-e270-11ea-96fc-f6a4490a2aae.png)
test_bc_nodx,docker
![test_bc_nodx](https://user-images.githubusercontent.com/41866052/90712217-94ddcd80-e270-11ea-8ad3-e881b9c809c7.png)
test_nobc_dx,docker
![test_nobc_dx](https://user-images.githubusercontent.com/41866052/90712234-99a28180-e270-11ea-9437-84ef281a5b62.png)
test_nobc_nodx,docker
![test_nobc_nodx](https://user-images.githubusercontent.com/41866052/90712239-9c9d7200-e270-11ea-967d-f901e33a885c.png)
test_nobc_nodx_noaln,docker
![test_nobc_nodx_noaln](https://user-images.githubusercontent.com/41866052/90712247-a1622600-e270-11ea-9416-8a76c371f5bf.png)

yuukiiwa committed:
Update check_samplesheet.py

yuukiiwa committed:
solve merge conflict with GoekeLab/nanoseq

yuukiiwa committed:
try solve merge conflicts with GoekeLab/nanoseq
